### PR TITLE
Mixed content mode issue fix

### PR DESCRIPTION
**Pull request by prabakarviji**
PR for fix - Image not displaying due to mixed content in android(#8460)

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
By analyzing the blame information on this pull request, we identified @mkonicek and @nicklockwood to be potential reviewers.

**Comment by charpeni** [comment]
I'm concerned about the security issue that pull request can do.

The WebView will attempt to be compatible with the approach of a modern web browser with regard to mixed content. Some insecure content may be allowed to be loaded by a secure origin and other types of content will be blocked. The types of content are allowed or blocked may change release to release and are not explicitly defined. This mode is intended to be used by apps that are not in control of the content that they render but desire to operate in a reasonably secure environment. For highest security, apps are recommended to use `MIXED_CONTENT_NEVER_ALLOW`.

IMO, this shouldn't be defined as `MIXED_CONTENT_COMPATIBILITY_MODE` by default but a prop will be interesting.

**Comment by ide** [comment]
Agree with the prop - let's have good defaults but also provide control.

File: ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java [hunk 1]
```diff
@@ -297,6 +293,13 @@ public void setInjectedJavaScript(WebView view, @Nullable String injectedJavaScr
     ((ReactWebView) view).setInjectedJavaScript(injectedJavaScript);
   }
 
+  @ReactProp(name = "enableMixedContentMode")
+  public void setMixedContentMode(WebView view, boolean enabled) {
+    if (enabled == true && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (ide) [inline ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java]: remove `== true`, just check `enabled`

File: Libraries/Components/WebView/WebView.android.js [hunk 2]
```diff
@@ -153,6 +153,12 @@ var WebView = React.createClass({
      * start playing. The default value is `false`.
      */
     mediaPlaybackRequiresUserAction: PropTypes.bool,
+    
+    /**
+     * Used on Android(5.0+), controls whether Mixed content mode is enabled or not
+     * @platform android
+     */
+    enableMixedContentMode: PropTypes.bool,
```
Comment (ide) [inline Libraries/Components/WebView/WebView.android.js]: can you rename this to mixedContentModeEnabled actually? other props in this component are named xxxEnabled

File: ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java [hunk 3]
```diff
@@ -293,7 +293,7 @@ public void setInjectedJavaScript(WebView view, @Nullable String injectedJavaScr
     ((ReactWebView) view).setInjectedJavaScript(injectedJavaScript);
   }
 
-  @ReactProp(name = "enableMixedContentMode")
+  @ReactProp(name = "mixedContentModeEnabled")
   public void setMixedContentMode(WebView view, boolean enabled) {
```
Comment (ide) [inline ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java]: and name this setMixedContentModeEnabled

**Comment by ide** [comment]
Looks good, thanks. Will merge after tests pass.

**Comment by prabakarviji** [comment]
@ide Updated the pull request.

File: ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java [hunk 4]
```diff
@@ -306,6 +307,13 @@ public void setInjectedJavaScript(WebView view, @Nullable String injectedJavaScr
     ((ReactWebView) view).setInjectedJavaScript(injectedJavaScript);
   }
 
+  @ReactProp(name = "mixedContentModeEnabled")
+  public void setMixedContentModeEnabled(WebView view, boolean enabled) {
+    if (enabled && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      view.getSettings().setMixedContentMode(WebSettings.MIXED_CONTENT_COMPATIBILITY_MODE);
```
Comment (ide) [inline ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java]: Sorry looking at the Android API more closely, I think we want to expose the underlying API: always, never, compatibility as enums.

**Comment by facebook-github-bot-3** [comment]
@prabakarviji do you have any updates for this pull request? It's been a while since the last update so wanted to check in and see if you've looked at the requested changes.

**Comment by mkonicek** [comment]
@prabakarviji I'm going through all pull requests and noticed this one hasn't been updated in a while and the last comment requests changes. I'll close this pull request so it doesn't stay open indefinitely but please send a new one if you want to continue working on this.

**Pull request closed without merging**